api/v2/checluster_types.go

```diff
@@ -541,6 +546,54 @@ type PodSecurityContext struct {
 	FsGroup *int64 `json:"fsGroup,omitempty"`
 }
 
+type CheClusterGitServices struct {
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitHub"
+	GitHub []GitHubService `json:"github,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitLab"
+	GitLab []GitLabService `json:"gitlab,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="BitBucket"
+	BitBucket []BitBucketService `json:"bitbucket,omitempty"`
+}
+
+// GitHubService enables users to work with a remote Git repository that is hosted on GitHub.
+type GitHubService struct {
+	// Kubernetes secret, that contains Base64-encoded GitHub OAuth Client id and GitHub OAuth Client secret data.
+	// The GitHub OAuth data must be stored in the Kubernetes secret in `id` and `secret` keys respectively.
```

Comment:
I think we should provide a link to the documentation. Not sure if markdown is rendered correctly though...
```suggestion
	// The GitHub OAuth data must be stored in the Kubernetes secret in `id` and `secret` keys respectively.
	// See [Eclipse Che Administration Guide](https://www.eclipse.org/che/docs/stable/administration-guide/configuring-oauth-2-for-github/) for details.
```